Uncertainty_Quantification/Deep500_recipe/loader.py

```diff
@@ -0,0 +1,253 @@
+    def __init__(self,
+                 args,
+                 sample_node: str,
+                 label_node: str,
+                 step="train",
+                 infer=False,
+                 year_dict=None):
+        super().__init__()
+        self.args = args
+        self.input_node = sample_node
+        self.label_node = label_node
+        self.index = 0
+        data_list = os.listdir(args.data_directory)
+        pat_x = ["inputLST" + i for i in year_dict[step]]
+        pat_y = ["inputLSTC" + i for i in year_dict[step]]
+        self.datalist_x = [
+            os.path.join(args.data_directory, i) for i in data_list
+            if any(j in i for j in pat_x)
+        ]
+        self.datalist_y = [
+            os.path.join(args.data_directory, i) for i in data_list
+            if any(j in i for j in pat_y)
+        ]
+        self.means = np.load(os.path.join(args.std_folder,
+                                          "means.npy")).astype(np.float32)
+        self.stddevs = np.load(os.path.join(args.std_folder,
+                                            "stddevs.npy")).astype(np.float32)
+        self.means = self.means[
+            2, None, :,  # args.parameters.index(args.pred_type),
+            :, :self.args.max_lat, :self.args.max_lon, ]
+        self.stddevs = self.stddevs[
+            2, None, :,  # args.parameters.index(args.pred_type),
+            :, :self.args.max_lat, :self.args.max_lon, ]
+        if args.dims == 2:
+            self.stddevs = self.stddevs[:, args.plvl_used, :, :]
+            self.means = self.means[:, args.plvl_used, :, :]
+        self.datalist_x.sort()
+        self.datalist_y.sort()
+
+        self.length = len(self.datalist_x)
+
+        if not infer:
+            assert self.length == len(
+                self.datalist_y
+            ), "number of input samples has to match the number of output samples"
+
+        self.infer = infer
+        self.step = step
+
+    def __len__(self):
+        return self.length
+
+    def __getitem__(self, idx):
+
+        data_x = np.load(self.datalist_x[idx])
+        data_x = reduce_sample_x(data_x, self.args, self.means, self.stddevs)
+
+        if not self.infer and self.step == "train":
+            data_y = np.load(self.datalist_y[idx])
+            data_y = reduce_sample_y(data_y, self.args)
+
+            for aug in self.args.augmentation:  # Apply transformations if chosen
+                data_x, data_y = TRANSFORMATION_DICTIONARY[aug](data_x, data_y,
+                                                                self.args)
+            return data_x.copy(), data_y.copy()
+        else:
+            if not self.infer and (self.step == "val" or self.step == "test"):
+                data_y = np.load(self.datalist_y[idx])
+                data_y = reduce_sample_y(data_y, self.args)
+
+                return data_x.copy(), data_y.copy()
+
+            return data_x.copy()
+
+
+class WeatherShuffleSampler(ShuffleSampler):
```

Comment:
why is this not the normal shuffle sampler?

Reply:
The sampler wants to access multiple indices here:

https://github.com/deep500/deep500/blob/34e93a46dea17513ac705bb4392b3514fa9d87c6/deep500/lv2/sampler.py#L144-L145

And this runs into an error. However, we can do it (by changing the sampler and/or dataset we defined).